core/src/main/java/co/uk/mommyheather/advancedbackups/core/backups/ThreadedBackup.java

```diff
@@ -1,532 +1,532 @@
+                dest.mkdirs();
+                int max = toBackup.size();
+                int index = 0;
+
+                if (!this.shutdown) {
+                    BackupStatusInstance instance = new BackupStatusInstance();
+                    instance.setAge(System.currentTimeMillis());
+                    instance.setMax(max);
+                    instance.setProgress(index);
+                    instance.setState(BackupStatusInstance.State.STARTED);
+                    BackupStatusInstance.setInstance(instance);
+                }
+                for (Path path : toBackup) {
+                    File out = new File(dest, path.toString());
+                    if (!out.getParentFile().exists()) {
+                        out.getParentFile().mkdirs();
+                    }
+                    Files.copy(new File(ABCore.worldDir.toString(), path.toString()).toPath(), out.toPath());
+                    index++;
+                    if (!this.shutdown) {
+                        BackupStatusInstance instance = new BackupStatusInstance();
+                        instance.setAge(System.currentTimeMillis());
+                        instance.setMax(max);
+                        instance.setProgress(index);
+                        instance.setState(BackupStatusInstance.State.STARTED);
+                        BackupStatusInstance.setInstance(instance);
+                    }
+                }
+                time = dest.lastModified();
+            }
+
+            //Finally, update + write the manifest
+            if (complete || toBackup.size() >= count) {
+                if (differential) {
+                    manifest.differential.setChainLength(0);
+                    manifest.differential.setLastBackup(new Date().getTime());
+                } else {
+                    manifest.incremental.setChainLength(0);
+                    manifest.incremental.setLastBackup(new Date().getTime());
+                }
+            } else {
+                if (differential) {
+                    manifest.differential.chainLength++;
+                    //manifest.differential.setLastBackup(new Date().getTime());
+                } else {
+                    manifest.incremental.chainLength++;
+                    manifest.incremental.setLastBackup(new Date().getTime());
+                }
+            }
+
+            FileWriter writer = new FileWriter(manifestFile);
+            writer.write(gson.toJson(manifest));
+            writer.flush();
+            writer.close();
+        } catch (IOException e) {
+            ABCore.logStackTrace(e);
+            throw e;
+        }
+
+
+    }
+
+    public void snapshot() {
+        this.snapshot = true;
+    }
+
+    public void shutdown() {
+        this.shutdown = true;
+    }
+
+
+    private String getFileHash(Path path) {
+        try {
+            MessageDigest md5 = MessageDigest.getInstance("MD5");
```

Comment:
Having ai offer to "fix" this for me feels... strange.
MD5 is preferred over SHA-256 here because it's faster, and security is not of concern here - the hash is just a way to see if the file has changed since the last backup and there's not a possible attack vector, nor is it used for any access control or similar.

Regardless, when the SPI system is implemented, this should be replaced with something more performant.